backend/src/main/java/com/example/subjecthub/controller/SubjectServiceController.java

```diff
@@ -101,7 +107,13 @@ public SubjectComment getComment(
         @PathVariable Long subjectId,
         @PathVariable Long commentId
     ){
-        return subjectCommentRepository.findBySubject_IdAndId(subjectId,commentId);
+        SubjectComment result;
+        try {
+            result = subjectCommentRepository.findBySubject_IdAndId(subjectId, commentId);
+        }catch(NullPointerException|ClassCastException e){
+            throw new SubjectHubException("Specified comment id: "+commentId+", not found for subject id: "+subjectId);
```

Comment:
As a tip, in the future I recommend using String.format() instead of adding strings together. It makes it troublesome to edit it in the future. So for example, if you needed to add the comment text and the subject name to the error, you'd have to edit it from
```java
// before
"Specified comment id: "+commentId+", not found for subject id: "+subjectId
// to
"Specified comment id: "+commentId+" with text: "+result.getPost()+", not found for subject id: "+subjectId+", " + subject.getName() + ".";
// Instead you could just have it be
String.format("Specified comment id: %s with text: %s, not found for subject: %s, %s.", commentId, result.getPost(), subjectId, subject.getName());
```

With String.format() you don't need to muck about with the plus symbols when you want to change the parameters and it's also much easier to read the error message.

Reply:
took your advice, using string.format now :dancer: